pkg/storaged/clevis-dialogs.jsx

```diff
@@ -20,40 +20,84 @@
 import cockpit from "cockpit";
 import React from "react"; React;
 
-import { dialog_open, TextInput, PassInput } from "./dialogx.jsx";
+import { dialog_open, SelectOne, TextInput, PassInput, CheckBox } from "./dialogx.jsx";
 
 const _ = cockpit.gettext;
 
 export function add(client, block) {
     dialog_open({ Title: _("Add network key"),
                   Fields: [
-                      TextInput("url", _("Key server address"),
-                                { validate: val => !val.length && _("Server address cannot be empty") }),
+                      SelectOne("method", _("Method"), { },
+                                [ { value: "tang",
+                                    title: _("\"tang\" Binding server") },
+                                  { value: "http",
+                                    title: _("\"http\" Key escrow") }
+                                ]),
+                      TextInput("http_url", _("URL"),
+                                { validate: val => {
+                                    if (val.length === 0)
+                                        return _("URL cannot be empty");
+                                    if (!val.startsWith("http:") && !val.startsWith("https:"))
+                                        return _("URL must start with either \"http:\" or \"https:\"");
+                                  },
+                                  visible: vals => vals.method == "http"
+                                }),
+                      CheckBox("allow_plain_http", _("Allow \"http://\" URL"),
+                               { visible: vals => vals.method == "http",
+                                 validate: (val, vals) => {
+                                     if (vals.http_url.startsWith("http:") && !val)
+                                         return _("This box must be checked to confirm that the key will be transported without HTTPS");
+                                 }
+                               }),
```

Comment:
This feels rather strange to me..

Reply:
It is strange.  I copied the "clevis http encrypt" configuration options pretty verbatim into the dialog.  The way we could do it is to just have a checkbox: "[ ] Use TLS" (on by default) and then adjust the URL accordingly.